internals/overlord/logstate/gatherer.go

```diff
@@ -250,6 +261,8 @@ func (g *logGatherer) Stop() {
 type timer struct {
 	timer *time.Timer
 	set   bool
+	// If non-nil, the timer won't expire until after this time
+	retryAfter *time.Time
```

Comment:
Isn't this timer already a "retry timer"? It's precisely separating out the moments in which we need to contact the client again, so it's unexpected to me at least that we need to retransform that "try timer" into a "retry timer", if you see what I mean?